format/boost_python/nexus_ext.cc

```diff
@@ -20,11 +20,34 @@
 namespace dxtbx { namespace format { namespace boost_python {
 
   using namespace boost::python;
+  template <typename T>
+  inline herr_t custom_read(hid_t, hid_t, hid_t, scitbx::af::versa<T, scitbx::af::flex_grid<> >);
+
+  /**
+   * Custom read function for integers
+   */
+  template<>
+  inline herr_t custom_read<int>(hid_t dataset_id, hid_t mem_space_id, hid_t file_space_id,
+                          scitbx::af::versa<int, scitbx::af::flex_grid<> > data) {
+    return H5Dread(
+      dataset_id, H5T_NATIVE_INT, mem_space_id, file_space_id, H5P_DEFAULT, &data[0]);
+  }
+
+  /**
+   * Custom read function for doubles
+   */
+  template<>
```

Comment:
If you are adding `double`, is it worth adding `float` at the same time for future proofing?